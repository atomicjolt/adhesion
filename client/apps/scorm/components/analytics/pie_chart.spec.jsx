import React from 'react';
import { shallow } from 'enzyme';
import Chart from './pie_chart';

describe('Scorm Analytics Chart', () => {

  let result;
  let props;
  describe('pie chart', () => {
    beforeEach(() => {
      props = {
        data: [],
        colors: [],
      };

      result = shallow(<Chart {...props} />);
    });

    it('matches the snapshot', () => {
      expect(result).toMatchSnapshot();
    });
  });
});
