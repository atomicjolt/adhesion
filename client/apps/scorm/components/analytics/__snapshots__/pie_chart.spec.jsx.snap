// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scorm Analytics Chart pie chart matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Chart
    colors={Array []}
    data={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <AnimationDecorator(Pie)
          animationBegin={400}
          animationDuration={1500}
          animationEasing="ease"
          cx="50%"
          cy="50%"
          data={Array []}
          endAngle={450}
          fill="#808080"
          innerRadius={0}
          isAnimationActive={true}
          label={true}
          labelLine={true}
          legendType="rect"
          minAngle={0}
          nameKey="name"
          outerRadius="80%"
          paddingAngle={0}
          startAngle={90}
          stroke="#fff"
          valueKey="value"
>
          
</AnimationDecorator(Pie)>,
        <Legend
          align="right"
          iconSize={14}
          layout="vertical"
          verticalAlign="top"
/>,
        <Tooltip
          active={false}
          animationDuration={400}
          animationEasing="ease"
          coordinate={
                    Object {
                              "x": 0,
                              "y": 0,
                            }
          }
          cursor={true}
          cursorStyle={Object {}}
          isAnimationActive={true}
          itemSorter={[Function]}
          itemStyle={Object {}}
          labelStyle={Object {}}
          offset={10}
          separator=" : "
          viewBox={
                    Object {
                              "x1": 0,
                              "x2": 0,
                              "y1": 0,
                              "y2": 0,
                            }
          }
          wrapperStyle={Object {}}
/>,
      ],
      "height": 300,
      "margin": Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      },
      "style": Object {},
      "width": 400,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animationBegin": 400,
          "animationDuration": 1500,
          "animationEasing": "ease",
          "children": Array [],
          "cx": "50%",
          "cy": "50%",
          "data": Array [],
          "endAngle": 450,
          "fill": "#808080",
          "innerRadius": 0,
          "isAnimationActive": true,
          "label": true,
          "labelLine": true,
          "legendType": "rect",
          "minAngle": 0,
          "nameKey": "name",
          "outerRadius": "80%",
          "paddingAngle": 0,
          "startAngle": 90,
          "stroke": "#fff",
          "valueKey": "value",
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "right",
          "iconSize": 14,
          "layout": "vertical",
          "verticalAlign": "top",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "animationDuration": 400,
          "animationEasing": "ease",
          "coordinate": Object {
            "x": 0,
            "y": 0,
          },
          "cursor": true,
          "cursorStyle": Object {},
          "isAnimationActive": true,
          "itemSorter": [Function],
          "itemStyle": Object {},
          "labelStyle": Object {},
          "offset": 10,
          "separator": " : ",
          "viewBox": Object {
            "x1": 0,
            "x2": 0,
            "y1": 0,
            "y2": 0,
          },
          "wrapperStyle": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <AnimationDecorator(Pie)
            animationBegin={400}
            animationDuration={1500}
            animationEasing="ease"
            cx="50%"
            cy="50%"
            data={Array []}
            endAngle={450}
            fill="#808080"
            innerRadius={0}
            isAnimationActive={true}
            label={true}
            labelLine={true}
            legendType="rect"
            minAngle={0}
            nameKey="name"
            outerRadius="80%"
            paddingAngle={0}
            startAngle={90}
            stroke="#fff"
            valueKey="value"
>
            
</AnimationDecorator(Pie)>,
          <Legend
            align="right"
            iconSize={14}
            layout="vertical"
            verticalAlign="top"
/>,
          <Tooltip
            active={false}
            animationDuration={400}
            animationEasing="ease"
            coordinate={
                        Object {
                                    "x": 0,
                                    "y": 0,
                                  }
            }
            cursor={true}
            cursorStyle={Object {}}
            isAnimationActive={true}
            itemSorter={[Function]}
            itemStyle={Object {}}
            labelStyle={Object {}}
            offset={10}
            separator=" : "
            viewBox={
                        Object {
                                    "x1": 0,
                                    "x2": 0,
                                    "y1": 0,
                                    "y2": 0,
                                  }
            }
            wrapperStyle={Object {}}
/>,
        ],
        "height": 300,
        "margin": Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
          "top": 0,
        },
        "style": Object {},
        "width": 400,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animationBegin": 400,
            "animationDuration": 1500,
            "animationEasing": "ease",
            "children": Array [],
            "cx": "50%",
            "cy": "50%",
            "data": Array [],
            "endAngle": 450,
            "fill": "#808080",
            "innerRadius": 0,
            "isAnimationActive": true,
            "label": true,
            "labelLine": true,
            "legendType": "rect",
            "minAngle": 0,
            "nameKey": "name",
            "outerRadius": "80%",
            "paddingAngle": 0,
            "startAngle": 90,
            "stroke": "#fff",
            "valueKey": "value",
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "right",
            "iconSize": 14,
            "layout": "vertical",
            "verticalAlign": "top",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "animationDuration": 400,
            "animationEasing": "ease",
            "coordinate": Object {
              "x": 0,
              "y": 0,
            },
            "cursor": true,
            "cursorStyle": Object {},
            "isAnimationActive": true,
            "itemSorter": [Function],
            "itemStyle": Object {},
            "labelStyle": Object {},
            "offset": 10,
            "separator": " : ",
            "viewBox": Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            },
            "wrapperStyle": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
